docs/Windows.md

```diff
@@ -1,43 +1,172 @@
 
-#### Building and running swift code
+_Note:_ Currently you also need `icu-63.1`.
+
+#### Building Swift code
+
+You should use a CMake project to build a Swift program. An example CMake project with support for mixed-language support is available at <https://github.com/compnerd/swift-cmake-demo>.
+
+For building such a CMake project the following paths have to be added to the PATH environment variable (adjust those paths to your actual installation folders):
+
+- `C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin`
+- `C:\Library\cmake\bin`
+- `C:\Library\icu-64.2\bin`
+
+_Note:_ Currently, `C:\Library\icu-63.1\bin` has to be added, too.
+
+You can choose to add those paths permanently to your PATH environment variable, or you formulate a Windows batch script that first adds those paths to the PATH environment variable and then calls the build commands which are listed below. Be sure to add those paths before other paths, so e.g. use
 
-1. To have the required Visual Studio toolchain available ensure you use the `x64 Native Tools Command Prompt for VS 2017` that came with Visual Studio. Search inside the Visual Studio application folder to find it.
-2. You need to add the folder containing `swiftc.exe` to the `PATH` environment variable to use the toolchain.
-3. You also need to add the folder containing the icu dlls to the `PATH` environment variable. 
+SET PATH=%SCRIPT_DIR%\cmake\bin;%PATH%
+
+in the batch script. For the path to CMake, adding it permanently to your PATH environment variable does not really help (try `cmake -version` inside the Native Tools Command Prompt), you would have to set it manually in each instance the Native Tools Command Prompt first, so a batch script might be a good idea.
+
+The following commands build the project mentioned above (including tests), to be executed inside the project directory after the PATH environment variable is set):
 
-This can be achieved by entering the following into the command line.
 ```cmd
-path %PATH%;C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin
-path %PATH%;C:\Library\icu-64.2\bin
+cmake -G Ninja -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=YES
+cd build
+ninja
+ninja test
 ```
 
-Alternatively, to make these changes persistent go to the Windows environment variables edit dialog found in `System Properties` > `Advanced` > `Environment Variables` and edit the path variable to add the following two new items.
+Of course, those commands have to be executed inside the Native Tools Command Prompt.
+
+In a Windows batch script, you might want to add  `|| goto :end` at the end of each of those commands plus an `:end` mark to ensure that this list of commands is interrupted as soon as an error occurs. A full batch script to build the project in `C:\tmp\swift-cmake-demo` then might look as follows (of course, that batch script has to be called from within the Native Tools Command Prompt):
```

Comment:
I don't think that we should be covering best practices for writing Windows batch files.  That is beyond the scope of this document.